### PR TITLE
BUG : compress_content_stream not readable in acrobat

### LINKED ISSUE #1654: Lost of "Page Mode" & preset zoom in "Reduce PDF Size" tutorials
Using https://pypdf.readthedocs.io/en/stable/user/file-size.html examples I experienced lost of "Page Mode" & "preset zoom" in output files

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python3 -m platform
macOS-10.10.5-x86_64-i386-64bit

$ python3 -c "import pypdf;print(pypdf.__version__)"
3.4.1
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
from pypdf import PdfReader, PdfWriter

dir=''

reader = PdfReader(dir+'bid.pdf')
writer = PdfWriter()

for page in reader.pages:
    page.compress_content_streams()  # This is CPU intensive!
    writer.add_page(page)

writer.add_metadata(reader.metadata)

with open(dir+'out.pdf', "wb") as f:
    writer.write(f)
```

Share here the PDF file(s) that cause the issue. The smaller they are, the
better. Let us know if we may add them to our tests!
Input file 
[input.pdf.zip](https://github.com/py-pdf/pypdf/files/10808515/bid.pdf.zip)
Output file
[out.pdf.zip](https://github.com/py-pdf/pypdf/files/10808550/out.pdf.zip)

OK to use these files for your tests

## Traceback

None

### PR DESCRIPTION
**Pull request by pubpub-zz**
fixes  #1654
ContentStream must be stored as individual objects

### CODE REVIEW

**Comment by pubpub-zz** [comment]
many change in the tests as the issue is coming from the fact that streams must be indirect_objects: compression can only be applied to pages part of PdfWriter

**Comment by pubpub-zz** [comment]
all good

File: tests/test_workflows.py [hunk 1]
```diff
@@ -489,44 +489,27 @@ def test_extract_text(url, name, strict, exception):
...
-def test_compress_raised(url, name):
-    data = BytesIO(get_pdf_from_url(url, name=name))
-    reader = PdfReader(data)
-    # no more error since iss #1090 fix
-    for page in reader.pages:
-        page.compress_content_streams()
-
-
-@pytest.mark.enable_socket()
-@pytest.mark.slow()
```
Comment (MartinThoma) [inline tests/test_workflows.py:503]: Note for myself: The slowest test now is the one with https://corpora.tika.apache.org/base/docs/govdocs1/950/950337.pdf-tika-950337.pdf with about 1.4s. We only want to mark tests with > 5s with `slow`, so removing this is fine.

**Comment by MartinThoma** [comment]
I've just noticed that the benchmark failed.

Did this PR make a breaking change or was the benchmark broken before?
Is the `raise ValueError("Page must be part of a PdfWriter")` actually correct?

**Pull request merged**